src/main/java/com/aws/greengrass/deployment/IotJobsHelper.java

```diff
@@ -150,6 +150,8 @@
 
     private AtomicBoolean isSubscribedToIotJobsTopics = new AtomicBoolean(false);
     private Future<?> subscriptionFuture;
+    private String thingName;
```

Comment:
`volatile` might be needed here? As multiple threads could write and read at the same time and a memory inconsistency might happen

Reply:
The variable is written only in the setupCommWithIotJobs method. Multiple threads cannot write to this, since I am making sure that when setupCommWithIotJobs is invoked, the previous invocation is cancelled.